packages/react-router/__tests__/navigate-test.tsx

```diff
@@ -115,4 +116,65 @@ describe("navigate", () => {
       expect(node.innerHTML).toMatchInlineSnapshot(`"<h1>About</h1>"`);
     });
   });
+
+  describe("with a search param", () => {
+    it("navigates to the correct URL with params", () => {
+      function Home() {
+        let navigate = useNavigate();
+
+        function handleClick() {
+          navigate({
+            pathname: "../about",
+            search: new URLSearchParams({ user: "mj" }).toString()
```

Comment:
Same comment as below about just using `"?user=mj"` as your `search` value.

Reply:
Same response, I was trying to write these as a user might (a la #7496) but if you feel strongly about it I can change them!